P/ProGraphMSA/build_tarballs.jl

```diff
@@ -0,0 +1,132 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "ProGraphMSA"
+version = v"2020.09.29"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/acg-team/ProGraphMSA.git", "51cf68a58f2e8900b3cfb40d24d335f7405abbed"),
+    ArchiveSource("https://sourceforge.net/projects/tclap/files/tclap-1.2.2.tar.gz", "f5013be7fcaafc69ba0ce2d1710f693f61e9c336b6292ae4f57554f59fde5837"),
+    DirectorySource(joinpath(@__DIR__, "patches"))
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+##########
+# 0) Environment
+##########
+ROOT="${WORKSPACE}/srcdir"
+PG_DIR="${ROOT}/ProGraphMSA"
+TCLAP_DIR="${ROOT}/tclap-1.2.2"
+
+# Disable Eigen vectorization/asserts for portability
+COMMON_EIGEN_DEFS="-DEIGEN_DONT_VECTORIZE -DEIGEN_DISABLE_UNALIGNED_ARRAY_ASSERT"
+
+# Prefer Ninja if available
+GEN="Unix Makefiles"
+command -v ninja >/dev/null 2>&1 && GEN="Ninja"
+
+##########
+# 1) Install TCLAP (header-only)
+##########
+mkdir -p "${prefix}/include/tclap"
+cp -av "${TCLAP_DIR}/include/tclap/"*.h "${prefix}/include/tclap/"
+
+##########
+# 2) Patch ProGraphMSA sources (Linux portability only)
+##########
+pushd "${PG_DIR}"
+
+# Strip problematic flags if present (SSE on non-x86, unsafe fast-math)
+for f in CMakeLists.txt src/CMakeLists.txt; do
+  if [ -f "$f" ]; then
+    sed -i -E 's/[[:space:]]*-msse2[[:space:]]*/ /g' "$f" || true
+    sed -i -E 's/[[:space:]]*-ffast-math[[:space:]]*/ /g' "$f" || true
+  fi
+done
+
+# Provide std::char_traits specializations so libc++ can handle custom alphabets.
+atomic_patch -p1 "${WORKSPACE}/srcdir/char_traits.patch"
+atomic_patch -p1 "${WORKSPACE}/srcdir/fix_codon_table.patch"
+
+popd
+
+##########
+# 3) Configure & build
+##########
+pushd "${PG_DIR}"
+
+# Eigen headers: typically provided by Eigen_jll at ${prefix}/include/eigen3
+EXTRA_INC="-I${prefix}/include -I${prefix}/include/eigen3"
+
+if [[ "${target}" == *-apple-darwin* ]]; then
+  sed -i "s/${target}-clang++/${target}-g++/g" "${CMAKE_TARGET_TOOLCHAIN}"
+  sed -i "s/${target}-gcc++/${target}-g++/g" "${CMAKE_TARGET_TOOLCHAIN}"
+  sed -i "s/${target}-clang/${target}-gcc/g" "${CMAKE_TARGET_TOOLCHAIN}"
+  sed -i "s/${target}-gccclang/${target}-gcc/g" "${CMAKE_TARGET_TOOLCHAIN}"
+fi
+
+rm -rf build
+cmake_extra=()
+if [[ "${target}" == *-apple-darwin* ]]; then
+  cmake_extra+=(
+    -DCMAKE_C_COMPILER="${target}-gcc"
+    -DCMAKE_CXX_COMPILER="${target}-g++"
+  )
+fi
```

Comment:
Why doesn't the Apple toolchain work, and why are there two sets of changes happening here?